spec/integration/messages/i18n_spec.rb

```diff
@@ -203,6 +214,26 @@ def store_errors(**errors)
           expect(meta).to eql(code: 123)
         end
       end
+
+      context "with meta-data and arg type" do
+        it "returns a template for a specific rule, meta and default arg type" do
+          store_errors(
+            predicate_with_meta: {
+              arg: {
+                default: {
+                  text: "text %{arg}",
+                  code: 123
+                }
+              }
+            }
+          )
+
+          template, meta = messages[:predicate_with_meta, path: :pages, locale: :en]
+
+          expect(template.(arg: "some arg")).to eql("text some arg")
+          expect(meta).to eq({code: 123})
+        end
+      end
```

Comment:
Initially I wanted to jest drop the logic around `text_key` in  `lib/dry/schema/messages/i18n.rb` - but it was making this possible and they might be people relying on this behavior so I thought it was worth to add test for it.